runtime/autoload/provider/python3.vim

```diff
@@ -1,45 +1,15 @@
-" The Python3 provider uses a Python3 host to emulate an environment for running
-" python3 plugins. :help provider
-"
-" Associating the plugin with the Python3 host is the first step because
-" plugins will be passed as command-line arguments
-
 if exists('g:loaded_python3_provider')
   finish
 endif
-let [s:prog, s:err] = provider#pythonx#Detect(3)
-let g:loaded_python3_provider = empty(s:prog) ? 1 : 2
 
-function! provider#python3#Prog() abort
-  return s:prog
+function! provider#python3#Call(method, args) abort
+  return v:lua.require'vim.provider.python'.call(a:method, a:args)
```

Comment:
```suggestion
  return v:lua.require('vim.provider.python').call(a:method, a:args)
```

It'd be a nit-picking, but can we use `v:lua.require(...)` instead (three similar ones)? The reason is that vimlsp complains these (E488: Trailing characters: ')

Reply:
That's a limitation of `v:lua`, see `:help v:lua-call`